File: src/main/java/org/parham/seasonjob/depend/Luckperms.java
package org.parham.seasonjob.depend;

import net.luckperms.api.LuckPerms;
import net.luckperms.api.LuckPermsProvider;
import net.luckperms.api.model.user.User;
import net.luckperms.api.node.Node;
import net.luckperms.api.node.NodeEqualityPredicate;
import org.bukkit.Bukkit;
import org.parham.seasonjob.SeasonJob;
import org.parham.seasonjob.data.job.Job;
import org.parham.seasonjob.data.job.JobManager;

import java.util.UUID;

public class Luckperms {
    private static int SYNC_INIT_TIME_SEC = 10;
    public static boolean isEnable = false;

    public static void checkEnabled() {
        if (SeasonJob.getInstance().getConfig().getBoolean("luck-perms")) {
            isEnable = Bukkit.getPluginManager().getPlugin("LuckPerms") != null;
        }
    }

    public static void updater() {
        Bukkit.getScheduler().runTaskTimerAsynchronously(SeasonJob.getInstance(), () -> {
            LuckPerms luckPerms = LuckPermsProvider.get();

            for (String jobName : JobManager.getJobsList()) {
                Job job = JobManager.getJob(jobName);

                if (!luckPerms.getGroupManager().isLoaded(jobName)) {
                    luckPerms.getGroupManager().createAndLoadGroup(jobName);
                }
                for (UUID id : job.getMembers()) {
                    Node jobRank = Node.builder("group." + jobName).build();
                    User user = luckPerms.getUserManager().getUser(id);

                    if (luckPerms.getUserManager().isLoaded(id)) {
                        user.data().add(jobRank);
                        LuckPermsProvider.get().getUserManager().saveUser(user);
                    }
                }
            }
        }, 1, 20 * SYNC_INIT_TIME_SEC);
    }
}

File: src/main/kotlin/Libs/API/ir/parham/SeasonJobsAPI/Dependencies/Luckperms.kt
package Libs.API.ir.parham.SeasonJobsAPI.Dependencies

import ir.parham.SeasonJobsAPI.Actions.Job
import net.luckperms.api.LuckPermsProvider
import net.luckperms.api.model.group.Group
import net.luckperms.api.model.user.User
import net.luckperms.api.node.Node
import org.bukkit.Bukkit
import java.util.UUID

class Luckperms {
    fun isEnable() : Boolean
    {
        if (Bukkit.getPluginManager().getPlugin("LuckPerms") != null)
        {
            return true
        }
        return false
    }
    fun createGroup(name: String) {
        if (!isEnable())
        {return}
        else
        {
            if (!LuckPermsProvider.get().groupManager.isLoaded(name))
            {
                LuckPermsProvider.get().groupManager.createAndLoadGroup(name)
            }
        }
    }
    fun deleteGroup(name: String) {
        if (!isEnable())
        {return}
        else
        {
            if (LuckPermsProvider.get().groupManager.isLoaded(name))
            {
                val group : Group? = LuckPermsProvider.get().groupManager.getGroup(name)

                if (group != null) {
                    LuckPermsProvider.get().groupManager.deleteGroup(group)
                }
            }
        }
    }
    fun addRank(uuid: UUID, name : String)
    {
        if (!isEnable())
        {return}
        else
        {
            if (LuckPermsProvider.get().groupManager.isLoaded(name)) {
                if (LuckPermsProvider.get().userManager.isLoaded(uuid)) {
                    val user: User = LuckPermsProvider.get().userManager.getUser(uuid)!!
                    user.data().add(Node.builder("group.${name}").build())

                    LuckPermsProvider.get().userManager.saveUser(user)
                }
            }
        }
    }
    fun removeRank(uuid: UUID, name : String)
    {
        if (!isEnable())
        {return}
        else
        {
            if (LuckPermsProvider.get().groupManager.isLoaded(name)) {
                if (LuckPermsProvider.get().userManager.isLoaded(uuid)) {
                    val user: User = LuckPermsProvider.get().userManager.getUser(uuid)!!
                    user.data().remove(Node.builder("group.${name}").build())

                    LuckPermsProvider.get().userManager.saveUser(user)
                }
            }
        }
    }
}